lib/svg_icon.js

```diff
@@ -0,0 +1,56 @@
+const xml2js = require('xml2js');
+const xmlBuilder = new xml2js.Builder()
+const util = require('util')
+const parseString = util.promisify(xml2js.parseString)
+const colorLib = require('color');
+
+const ignoreColors = ['none', '#FFF', '#fff', '#ffffff', '#FFFFFF'];
+const TRANSPORT_BLUE = '#123295';
+
+module.exports.parseIcon = async function(svgStream) {
+  let data = await parseString(svgStream)
+
+  let color = (data.svg.rect || [])
+    .map(rectNode => rectNode['$'].fill)
+    .filter(color => color && !ignoreColors.includes(color))[0];
+  if (!color) {
+    color = TRANSPORT_BLUE;
+  }
```

Comment:
We could look for the color of the `<path>` as a fallback instead.  
But I suppose this logic should be removed with a future rewriting, so maybe it's not necessary to worry about this now.

Reply:
I'll see if I can make a generic process while trying to re-introduce currently ignored icons.